modules/nf-core/gatk4/printreads/main.nf

```diff
@@ -0,0 +1,60 @@
+process GATK4_PRINTREADS {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda (params.enable_conda ? "bioconda::gatk4=4.3.0.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/gatk4:4.3.0.0--py36hdfd78af_0':
+        'quay.io/biocontainers/gatk4:4.3.0.0--py36hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(input), path(index)
+    tuple val(meta2), path(fasta)
+    path (fai)
+    path (dict)
+
+    output:
+    tuple val(meta), path("*.reads.bam") , emit: bam,   optional: true
+    tuple val(meta), path("*.reads.cram"), emit: cram,  optional: true
+    tuple val(meta), path("*.reads.sam") , emit: sam,   optional: true
+    path "versions.yml"                  , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def avail_mem = 3
+    if (!task.memory) {
+        log.info '[GATK PrintReads] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
+    } else {
+        avail_mem = task.memory.giga
+    }
+    """
+    gatk --java-options "-Xmx${avail_mem}g" PrintReads \\
+        $args \\
+        --reference $fasta \\
+        --input $input \\
+        --read-index $index \\
+        --output ${prefix}.reads.${input.getExtension()}
```

Comment:
It's preferred to not put any custom extensions on the output files. The pipeline creator should add this themselves using `ext.prefix`. You could also add a check to see if the output file has the same name as the input file and if so, return an error (see [here](https://github.com/nf-core/modules/blob/3dd73937b084b547f9272bc901e0f120a7913fd8/modules/nf-core/bedtools/closest/main.nf#L31-L33))

Reply:
That's pretty neat! TIL! 

And btw, does bedtools/closest module throw an error for you when the input and output filenames are the same? I thought we had to use string interpolation (like `"${input_1}"` and `"${input_2}"`) to get the values of the input variables in the script block.. 🤔

Reply:
I haven't actually tested this tbh :grin: I copied that piece of code from another module. But you're probably right about it